invenio_rdm_records/records/dumpers.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite-based data model for Invenio."""
+
+import calendar
+from datetime import date
+
+from edtf import parse_edtf
+from edtf.parser.edtf_exceptions import EDTFParseException
+from invenio_records.dumpers import ElasticsearchDumperExt
+
+
+class EDTFDumperExt(ElasticsearchDumperExt):
+    """Elasticsearch dumper extension for EDTF dates support.
+
+    It adds two fields: publication_date_start and publication_date_end,
+    which correspond to the lower strict and upper strict bounds. This
+    are required for sorting. In an ascending sort (most recent last),
+    sort by publication_date_start to get a natural sort order. In a
+    descending sort (most recent first), sort by publication_date_end.
+    """
+
+    def __init__(self, field):
+        """Constructor.
+
+        :param field: dot separated path to the EDTF field to process.
+        """
+        super(EDTFDumperExt, self).__init__()
+        self.field_path = field.split('.')
+
+    def _dump(self, path, data):
+        """Find recursively and dump EDTF field."""
+        key = path[0]
+        try:
+            if len(path) > 1:
+                data[key] = self._dump(path[1:], data[key])
+            else:
+                pd = parse_edtf(data[key])
+                data[f"{key}_start"] = date.fromtimestamp(
+                    calendar.timegm(pd.lower_strict())).isoformat()
+                data[f"{key}_end"] = date.fromtimestamp(
+                    calendar.timegm(pd.upper_strict())).isoformat()
+
+            return data
+        except (KeyError, EDTFParseException):
+            # The field does not exists or had wrong data
+            return data  # FIXME: should log this in debug mode?
+
+    def dump(self, record, data):
+        """Dump the data."""
+        self._dump(self.field_path, data)
+
+    def _load(self, path, data):
+        """Find recursively and remove the dump fields."""
+        key = path[0]
+
+        if len(path) > 1:
+            data[key] = self._load(path[1:], data[key])
```

Comment:
```python
keys = parse_lookup_key(self.field_path)
key = keys[-1]
data = dict_lookup(data, keys, parent=True)
data.pop(f'{key}_start', None)
...
```

Reply:
Implemented with the restrictions shown above.